x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_creation/components/runtime_mappings/switch_modal.tsx

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC } from 'react';
+import { EuiConfirmModal } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+interface Props {
+  onCancel: () => void;
+  onConfirm: () => void;
+  confirmButtonDisabled?: boolean;
+}
+
+const modalTitle = i18n.translate(
+  'xpack.ml.dataframe.analytics.createWizard.runtimeEditorSwitchModalTitle',
+  {
+    defaultMessage: 'Edits will be lost',
+  }
+);
+
+const cancelButtonText = i18n.translate(
+  'xpack.ml.dataframe.analytics.createWizard.runtimeEditorSwitchModalCancelButtonText',
+  {
+    defaultMessage: 'Cancel',
+  }
+);
+
+const applyChangesText = i18n.translate(
+  'xpack.ml.dataframe.analytics.createWizard.runtimeEditorSwitchModalConfirmButtonText',
+  {
+    defaultMessage: 'Apply changes',
+  }
+);
+const modalMessage = i18n.translate(
+  'xpack.ml.dataframe.analytics.createWizard.runtimeEditorSwitchModalBodyText',
+  {
+    defaultMessage: `The changes in the editor haven't been applied yet. Apply changes to avoid losing your edits.`,
+  }
+);
+
+export const SwitchModal: FC<Props> = ({ onCancel, onConfirm, confirmButtonDisabled = false }) => (
```

Comment:
I wonder if the choices in the modal need changing, as there doesn't seem to be any way of discarding edits in the editor for runtime fields:

![image](https://user-images.githubusercontent.com/7405507/114859734-10355600-9de3-11eb-8396-ef130050c817.png)

You can hit Apply changes to apply the edits and close the editor, and hitting Cancel will keep the editor open but with the existing edits. Compare this to the choices in the modal for the Transforms query:

![image](https://user-images.githubusercontent.com/7405507/114859977-58547880-9de3-11eb-8250-c6364aa49ca2.png)

The `Switch to query bar` button gives the user of discarding their edits, which I think is required functionality.